### DISCUSSION TITLE
Is there a Plugin support for Flux?

### DISCUSSION THREAD

**Discussion by aabedraba** [fluxcd/flux2#2762] [Q&A] [answered]
Hi all, 

Coming from ArgoCD, I had the option to run [ConfigPluginManagement](https://argo-cd.readthedocs.io/en/stable/user-guide/config-management-plugins/) which allows you to give ArgoCD a binary (that'll be the plugin) and then that binary would take a file format and convert it to a manifest which would then be applied and synced by ArgoCD.

I have tried to find the equivalent of that in Flux, but the closest thing I found was [source-controller](https://fluxcd.io/docs/gitops-toolkit/source-watcher/)(?) where if I wanted to have the same flow as before, I'd be running my binary from the go program and then I'd need to publish the changes to a git server so Flux can pick it up?

Is there another alternative to run plugins in Flux?

**Comment by hiddeco** [comment, marked as answer]
There is not, and I do not think there ever will be. One of our philosophies is to absolutely limit the number of binaries and shelling out, as our experiences with Flux v1 have been that arbitrary shell-outs are hard to debug and provide support for, in addition to it being generally tricky in terms of security.

Our opinion and vision is that with support for OCI that will be added soon, this should cover for most cases. As you can then move the generating bit to CI and publish it as an OCI artifact, which can then be picked up by the source-controller.

**Reply by hiddeco** [reply]
- RFC: https://github.com/fluxcd/flux2/pull/2601
- PoC implementation: https://github.com/fluxcd/flux2/pull/2601

**Reply by aabedraba** [reply]
Thanks so much for your (super quick!!) response @hiddeco!
